notebooks/HADDOCK3-antibody-antigen.ipynb

```diff
@@ -0,0 +1,2834 @@
+        "\n",
+        "\n",
+        "---\n",
+        "\n",
+        "\n",
+        "## HADDOCK3 workflow definition\n",
+        "\n",
+        "The first step is to create a HADDOCK3 configuration file that will define the docking workflow.\n",
+        "We will follow a classic HADDOCK workflow consisting of rigid body docking, semi-flexible refinement and final energy minimisation followed by clustering.\n",
+        "\n",
+        "We will also integrate two analysis modules in our workflow:\n",
+        "\n",
+        "- `caprieval` will be used at various stages to compare models to either the best scoring model (if no reference is given) or a reference structure, which in our case we have at hand (`pdbs/4G6M_matched.pdb`). This will directly allow us to assess the performance of the protocol. In the absence of a reference, `caprieval` is still usefull to assess the convergence of a run and analyse the results.\n",
+        "- `contactmap` added as last module will generate contact matrices of both intra- and intermolecular contacts and a chordchart of intermolecular contacts for each cluster.\n",
+        "\n",
+        "\n",
+        "Our workflow consists of the following modules:\n",
+        "\n",
+        "1. **`topoaa`**: *Generates the topologies for the CNS engine and builds missing atoms*\n",
+        "2. **`rigidbody`**: *Performs rigid body energy minimisation (`it0` in haddock2.x)*\n",
+        "3. **`caprieval`**: *Calculates CAPRI metrics (i-RMSD, l-RMSD, Fnat, DockQ) with respect to the top scoring model or reference structure if provided*\n",
+        "4. **`seletop`** : *Selects the top X models from the previous module*\n",
+        "5. **`flexref`**: *Preforms semi-flexible refinement of the interface (`it1` in haddock2.4)*\n",
+        "6. **`caprieval`**\n",
+        "7. **`emref`**: *Final refinement by energy minimisation (`itw` EM only in haddock2.4)*\n",
+        "8. **`caprieval`**\n",
+        "9. **`clustfcc`**: *Clustering of models based on the fraction of common contacts (FCC)*\n",
+        "10. **`seletopclusts`**: *Selects the top models of all clusters*\n",
+        "11. **`caprieval`**\n",
+        "12. **`contactmap`**: *Contacts matrix and a chordchart of intermolecular contacts*\n",
+        "\n",
+        "\n",
+        "The corresponding toml configuration file (provided in `workflows/docking-antibody-antigen.cfg`) looks like:\n",
+        "\n",
+        "```\n",
+        "# ====================================================================\n",
+        "# Antibody-antigen docking example with restraints from the antibody\n",
+        "# paratope to the NMR-identified epitope on the antigen\n",
+        "# ====================================================================\n",
+        "\n",
+        "# Directory in which the scoring will be done\n",
+        "run_dir = \"run1\"\n",
+        "\n",
+        "# Compute mode\n",
+        "mode = \"local\"\n",
+        "ncores = 50\n",
```

Comment:
keep in mind this will most likely exhaust all processors of a notebook and render the device unusable during the execution of this notebook

Reply:
haddock3 is limited by the number of available core on the system. So if only two available it will use max 2.